ciphers/permutation_cipher.py

```diff
@@ -0,0 +1,139 @@
+"""
+The Permutation Cipher, implemented above, is a simple encryption
+technique that rearranges the characters in a message based on a secret key.
+It divides the message into blocks and applies a permutation to the characters
+within each block according to the key. The key is a sequence of unique integers
+that determine the order of character rearrangement. For more info read:-
+https://www.nku.edu/~christensen/1402%20permutation%20ciphers.pdf
+
+"""
+import random
+
+
+def generate_valid_block_size(message_length: int) -> int:
+    """
+    Generate a valid block size that is a factor of the message length.
+
+    Args:
+        message_length (int): The length of the message.
+
+    Returns:
+        int: A valid block size.
+
+    Example:
+        >>> generate_valid_block_size(2)
+        2
+    """
+    while True:
+        block_size = random.randint(2, message_length)
+        if message_length % block_size == 0:
+            return block_size
```

Comment:
You could use `random.choice` to select a single random element from a list:
```suggestion
    block_sizes = [
        block_size for block_size in range(2, message_length + 1)
        if message_length % block_size == 0
    ]
    return random.choice(block_size)
```
This would avoid having to call `random.randint` repeatedly. However, I'm not sure that this would actually be more efficient, so it's up to you.